### ISSUE TITLE
pubsub/query: QueryParser.Init's logic is complex but not sufficiently tested

### ISSUE THREAD

**Issue by odeke-em** [tendermint/tendermint#5909]
I am flagging this as a vector for eyes to check it out. I am also going to fuzz it tonight. The logic for the query parser is complex and not fully tested, given the recursive descent parser that it is.

**Comment by tessr** [comment]
@odeke-em did your fuzzing reveal any issues?

**Comment by odeke-em** [comment]
Howdy Tess! I got side tracked for a bit, but this weekend I shall run it.
Thanks for the reminder.

**Comment by odeke-em** [comment]
Sorry for the late reply @tessr! I did fuzz it for a while but no reports of crashes so far. Also I just noticed that there is a fuzz test in the same directory, the only problem is the lack of corpora https://github.com/tendermint/tendermint/tree/master/libs/pubsub/query/fuzz_test. One thing going forward to consider would be to submit continuous fuzzing to oss-fuzz https://google.github.io/oss-fuzz/

We can now close this issue!
